### PR TITLE
Aggregregate metrics on all branches

### PR DESCRIPTION
**Pull request by mgrauer**
@cpatrick PTAL

This includes all the commits of #202.  I can give you a demo tomorrow.

The behavior change is to split the aggregate metrics from notifications.  Metrics defined by an aggregate metric spec will be calculated on all submissions.  Notifications will only be set by an aggregate metric notification, which defines a threshold and a branch, and can have users linked to it who will be notified when a metric is over threshold.

### CODE REVIEW

**Comment by mgrauer** [comment]
One more TODO, test that notification jobs are correctly scheduled when aggregate metrics are computed on a submission via the ApiComponent.

File: modules/tracker/controllers/components/ApiaggregatemetricnotificationComponent.php [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+
+        /** @var Tracker_AggregateMetricNotificationModel $aggregateMetricNotificationModel */
+        $aggregateMetricNotificationModel = MidasLoader::loadModel('AggregateMetricNotification', $this->moduleName);
+        /** @var Tracker_AggregateMetricSpecModel $aggregateMetricSpecModel */
+        $aggregateMetricSpecModel = MidasLoader::loadModel('AggregateMetricSpec', $this->moduleName);
+
+        /** @var Tracker_AggregateMetricNotificationDao $aggregateMetricNotificationDao */
+        $aggregateMetricNotificationDao = $aggregateMetricNotificationModel->load($aggregateMetricNotificationId);
+        $userDao = $apihelperComponent->getUser($args);
+        if ($aggregateMetricSpecModel->policyCheck($aggregateMetricNotificationDao->getAggregateMetricSpec(), $userDao, MIDAS_POLICY_WRITE) === false) {
```
Comment (cpatrick) [inline modules/tracker/controllers/components/ApiaggregatemetricnotificationComponent.php]: MIDAS_POLICY_READ

Comment (mgrauer) [inline modules/tracker/controllers/components/ApiaggregatemetricnotificationComponent.php]: Here's where the READ policy change should be

**Comment by mgrauer** [comment]
@cpatrick 

GET now requires READ

**Comment by mgrauer** [comment]
This should be ok to merge from a conflict perspective.

**Comment by mgrauer** [comment]
Closing this PR as all new commits were merged in #231.

**Pull request merged**